business/health.go

```diff
@@ -222,6 +222,50 @@ func (in *HealthService) getNamespaceAppHealth(namespace string, appEntities nam
 	return allHealth
 }
 
+// GetNamespaceServiceHealth returns a health for all services in given Namespace (thus, it fetches data from K8S and Prometheus)
+func (in *HealthService) GetNamespaceServiceHealth(namespace, rateInterval string) (models.NamespaceServiceHealth, error) {
+	services, err := in.k8s.GetServices(namespace, nil)
+	if err != nil {
+		return nil, err
+	}
+	return in.getNamespaceServiceHealth(namespace, services, rateInterval), nil
+}
+
+func (in *HealthService) getNamespaceServiceHealth(namespace string, services []v1.Service, rateInterval string) models.NamespaceServiceHealth {
+	allHealth := make(models.NamespaceServiceHealth)
+
+	// Prepare all data
+	for _, service := range services {
+		h := models.ServiceHealth{}
+		allHealth[service.Name] = &h
+	}
+
+	// Fetch services requests rates
+	rates, _ := in.prom.GetAllRequestRates(namespace, rateInterval)
```

Comment:
+1, no need to make two queries here.  I'll add something.  By the way, I think the inbound/outbound terminology isn't quite right in GetAllRequestRates.  It's more like external (requests originating outside the ns)/internal (requests originating inside the namespace).  When I think inbound/outbound I think about traffic coming into and out of a workload.

One other note.  I wanted to mention that requests originating from "unknown" are not included in GetAllRequestRates.  This is because they are not reported with source telemetry.  I didn't know if this was intentional for health calculation, or an oversight.  To be consistent I will keep it the same for my new method.   This approach removes a query from the health methods, which is good, and may not be missed as we expect most systems to use gateways.  The graph does currently include the "unknown" traffic, and that may be sufficient.  In the future we may even want to make it optional.

Reply:
Can you look at my PR from today, it's exactly about that topic: #552 (though because of a different bug)